### DISCUSSION TITLE
After Update: conf.yml deleted - How to import & Warnings?

### DISCUSSION THREAD

**Discussion by Seelenkontrast** [Lissy93/dashy#572] [Help and Support] [answered]
Hi there,
sorry for this obvious question, but I use Dashy a few months now, and after every update, I have to configure the whole thing from scratch. I exported the config-file, but I can't import it.

I also don't know, where to put it after an update. I googled, but I couldn't figure it out.

My Setup: 

- Debian 11
- Portainer CE
- Dashy within Portainer

I also tried it via docker-compose, but there is the same problem. I hope you can help a newbie like me.

At every update I get two warnings, that say:

 WARN  A new version of sass-loader is available. Please upgrade for best experience.
 WARNING  Compiled with 2 warnings8:52:39 AM
 warning  
asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  fonts/Raleway-Variable.3b5635ac.ttf (302 KiB)
  js/chunk-vendors.282673aa.js (5.52 MiB)
  js/dashy.98fb8cf9.js (455 KiB)
  web-icons/screenshots/dashy-scrsht-1.png (367 KiB)
  web-icons/windows10/LargeTile.scale-400.png (307 KiB)
  web-icons/windows10/SplashScreen.scale-400.png (307 KiB)
 warning  
entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.

Entrypoints:
  dashy (6.15 MiB)
      css/chunk-vendors.d8067ad8.css
      js/chunk-vendors.282673aa.js
      css/dashy.6bdf0979.css
      js/dashy.98fb8cf9.js

How can I fix this? 
Thank you!

**Comment by lissy93** [comment, marked as answer]
You should just be able to paste your config into the conf.yml file, and map that to a Docker volume. Config shouldn't be reset from updates if it's in a volume, are you using watchtower or just recreating the container with docker-compose?

I recommend keeping an up-to-date backup of your config in YAML format, you can download this under the Export Config menu.

Also, take a look at the **[Management Docs](https://github.com/Lissy93/dashy/blob/master/docs/management.md)** - there are lots of resources about running, updating, backing up, etc, which should provide a good starting point. But generally Dashy is pretty standard, so any Docker info you learn elsewhere will apply. 

And those messages are [nothing to worry about](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#warnings-in-the-console-during-deploy). The size warning is from PWA assets which will only be loaded if you are using the PWA. And the latest version of sass-loader isn't supported by Vue CLI.

**Reply by Seelenkontrast** [reply]
Hi,
thanks for your fast answer. I would paste it, but I'm too dumb to locate the original conf.yml-file.
I backupped the actual config file before updating...

Ah and I'm recreating it via Docker-Compose. I tried it via docker-compose and recreation in Portainer.

**Comment by MattW42** [comment]
Hope this is of some help, as I went through similiar experience using Portainer.
I believe this way works when updating and the conf file is kept as persistant.

(The other option is to use the cloud backup and just restore the config each time )

You might need to tweak some of the host steps (step 2) to suit Debian, I was using a Raspberry Pi with latest Pi OS image.

I used the following steps to get this working in Portainer.

1. Create the '`dashy_data`' volume in Portainer

2. Create conf.yml (on host) and populate with default config

```
#  SSH to Pi - logon to host (in this case it was a RaspberryPi running docker)
      cd /var/lib/
#  Create conf.yml
      sudo touch docker/volumes/dashy_data/_data/conf.yml
#  Edit conf.yml - requires some basic config to start with
      sudo nano docker/volumes/dashy_data/_data/conf.yml

#  Full path to docker data - requires sudo otherwise access denied.
#    /var/lib/docker/volumes/dashy_data/_data/conf.yml

#  get full path name for stack info with 
      pwd

#  get uid and gid for stack info with
       id
```

3. Create a new stack (in Portainer) with the following yml config

```
---
version: "3.8"
volumes:
  dashy_data:

services:
  dashy:
    image: lissy93/dashy
    container_name: Dashy
    volumes:
      - dashy_data:/app/public
      
      # previous method for mapping to a file
      #- /var/lib/docker/volumes/dashy_data/_data/conf.yml:/app/public/conf.yml
    
    ports:
      - 80:80
    environment:
      - NODE_ENV=production
      - UID=1000
      - GID=1000
    restart: unless-stopped
```

4.  Deploy stack in Portainer ( Note: Portainer now shows the dashy_data volume in use - under the 'Volumes' menu )


For updating Dashy, you can either re-deploy the stack (there's an option to pull the latest image)
or you can install Watchtower in another container and automate the update process.
